packages/reapit-connect-management/src/components/ui/apps/app-installation-manager.tsx

```diff
@@ -0,0 +1,206 @@
+import { AppSummaryModel, InstallationModelPagedResult } from '@reapit/foundations-ts-definitions'
+import { Section, H5, FlexContainerBasic, Button } from '@reapit/elements'
+import useSWR from 'swr'
+import AppInstallationPerOfficeGroup from './app-installation-per-office-group'
+import AppInstallationConfirmationModal from './app-installation-confirmation-modal'
+import { URLS } from '../../../constants/api'
+import { clientIdEffectHandler } from '../../../utils/client-id-effect-handler'
+import { bulkInstall } from '../../../services/installation'
+import { notification } from '@reapit/elements'
+
+export interface AppInstallationManagerProps {
+  app: AppSummaryModel
+}
+
+export const SPECIFIC_OFFICE_GROUPS = 'SPECIFIC_OFFICE_GROUPS'
+export const WHOLE_ORG = 'WHOLE_ORG'
+
+const AppInstallationManager: React.FC<AppInstallationManagerProps> = ({ app }: AppInstallationManagerProps) => {
+  const [initialAppInstallationType, setInitialAppInstallationType] = useState<string>('')
+  const [appInstallationType, setAppInstallationType] = useState<string>('')
+  const [officeGroupsToAdd, setOfficeGroupsToAdd] = useState<string[]>([])
+  const [officeGroupsToRemove, setOfficeGroupsToRemove] = useState<string[]>([])
+  const [clientId, setClientId] = useState<string | null>(null)
+  const [showConfirmModal, setShowConfirmModal] = useState<boolean>(false)
+
+  useEffect(clientIdEffectHandler(clientId, setClientId), [])
+
+  const { data: installations, isValidating: installationsValidating, revalidate: revalidateInstallations } = useSWR<
+    InstallationModelPagedResult
+  >(`${URLS.INSTALLATIONS}/?AppId=${app.id}&IsInstalled=true&pageSize=999`)
+
+  if (clientId && installations?.data && !initialAppInstallationType) {
+    const clientIdFirstPart = clientId.split('-')[0]
+    const installedForWholeOrg = installations.data.some(installation => installation.client === clientIdFirstPart)
+    const installedForOfficeGroups = installations.data.some(installation =>
+      installation?.client?.startsWith(`${clientIdFirstPart}-`),
+    )
+
+    if (installedForWholeOrg) {
+      setInitialAppInstallationType(WHOLE_ORG)
+      setAppInstallationType(WHOLE_ORG)
+    } else if (installedForOfficeGroups) {
+      setInitialAppInstallationType(SPECIFIC_OFFICE_GROUPS)
+      setAppInstallationType(SPECIFIC_OFFICE_GROUPS)
+    }
+  }
+
+  const submitButtonEnabled =
+    appInstallationType === SPECIFIC_OFFICE_GROUPS
+      ? !!officeGroupsToAdd.length || !!officeGroupsToRemove.length
+      : appInstallationType !== initialAppInstallationType
+
+  const handleOnCheckboxChange = (
+    setAppInstallationType: Dispatch<SetStateAction<string>>,
+    appInstallationType: string,
+  ) => async () => {
+    setAppInstallationType(appInstallationType)
+    if (appInstallationType === WHOLE_ORG) {
+      setOfficeGroupsToAdd([])
+      setOfficeGroupsToRemove([])
+    }
+  }
+
+  const handleInstallConfirmation = async () => {
```

Comment:
I appreciate you haven't added tests yet but when you do, would be great if these handers were broken out of the component and tested - a fair amount of logic in there